actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
```

Comment:
@sb8244: Yes, I deleted my suggestion because `super` can cause the action to be performed. It needs to be called before the `performed?` check.